src/indent.cpp

```diff
@@ -1560,56 +1561,57 @@ void indent_text(void)
          frm.pse[frm.pse_tos].indent_tmp = frm.pse[frm.pse_tos].indent;
          frm.pse[frm.pse_tos].indent_tab = frm.pse[frm.pse_tos].indent;
          log_indent_tmp();
-
          indent_column_set(frm.pse[frm.pse_tos].indent_tmp);
 
-         if ((cpd.settings[UO_indent_class_colon].b && (pc->type == CT_CLASS_COLON)) ||
-             (cpd.settings[UO_indent_constr_colon].b && (pc->type == CT_CONSTR_COLON)))
+         if (cpd.settings[UO_indent_class_colon].b && (pc->type == CT_CLASS_COLON))
+         {
+            if (cpd.settings[UO_indent_class_on_colon].b)
+            {
+               frm.pse[frm.pse_tos].indent = pc->column;
+               log_indent();
+            }
+            else
+            {
+               next = chunk_get_next(pc);
+               if ((next != nullptr) && !chunk_is_newline(next))
+               {
+                  frm.pse[frm.pse_tos].indent = next->column;
+                  log_indent();
+               }
+            }
+         }
+         else if (cpd.settings[UO_indent_constr_colon].b && (pc->type == CT_CONSTR_COLON))
          {
             prev = chunk_get_prev(pc);
             if (chunk_is_newline(prev))
             {
                frm.pse[frm.pse_tos].indent += cpd.settings[UO_indent_ctor_init_leading].u;
                log_indent();
+            }
 
-               if (cpd.settings[UO_indent_ctor_init].u != 0)
-               {
-                  frm.pse[frm.pse_tos].indent += cpd.settings[UO_indent_ctor_init].u;
-                  log_indent();
-                  frm.pse[frm.pse_tos].indent_tmp += cpd.settings[UO_indent_ctor_init].u;
-                  frm.pse[frm.pse_tos].indent_tab += cpd.settings[UO_indent_ctor_init].u;
-                  log_indent_tmp();
-                  indent_column_set(frm.pse[frm.pse_tos].indent_tmp);
-               }
+            // TODO: Create a dedicated indent_constr_on_colon?
+            if (cpd.settings[UO_indent_class_on_colon].b)
+            {
+               frm.pse[frm.pse_tos].indent = pc->column;
+               log_indent();
+            }
+            else if (cpd.settings[UO_indent_ctor_init].n != 0)
+            {
+               frm.pse[frm.pse_tos].indent = std::max<ptrdiff_t>(frm.pse[frm.pse_tos].indent + cpd.settings[UO_indent_ctor_init].n, 0);
```

Comment:
I understand now. It would be better to have this comment inside the code because it's not really obvious.